build/blog.ts

```diff
@@ -42,12 +46,24 @@ function calculateReadTime(copy: string): number {
 }
 
 async function readPost(
-  file: string
+  file: string,
+  extraMeta = true
 ): Promise<{ blogMeta: BlogPostFrontmatter; body: string }> {
   const raw = await fs.readFile(file, "utf-8");
 
   const { attributes, body } = frontmatter<BlogPostFrontmatter>(raw);
   const readTime = calculateReadTime(body);
+
+  if (extraMeta) {
+    const posts = await allPostFrontmatter();
```

Comment:
🔴  We should avoid O(n^2) here. For each post, we read all posts.

Reply:
I had thought about it, but considered it a premature optimisation at this point considering few posts we have (and pretty slow growth ~2/week). I'll just slap em all in a global object I guess :)

Reply:
Actually I think I can just chuck a cheeky `memoize` on `allPostFrontmatter`, no?

Reply:
I disagree that avoiding O(n^2) is premature optimization. We're adding about one blog post per week, which makes 50*50=2500 read posts in 1 year.

As for `memoize`, I'd memoize the parameterized calls to `readPost(path: string)` instead.

Reply:
Having thought about it a bit, I reckon `allPostFrontmatter` should be the one memoized: if we memoize `readPost` we store the entire content of each post as well as its metadata in memory, whereas if we memoize `allPostFrontmatter` we just store the content of all metadata in memory.

I think the number of reads is the same across both approaches - so I'd go with memoizing `allPostFrontmatter` to keep memory usage lower.

Reply:
Sounds good.